interfaces/builtin/hardware_observe.go

```diff
@@ -31,22 +31,13 @@ const hardwareObserveConnectedPlugAppArmor = `
 
 #include <abstractions/base>
 
-/etc/udev/udev.conf r,
-@{PROC}/*/stat r,
-/run/udev/data/* r,
-/sys/bus/ r,
-/sys/bus/**/ r,
-/sys/class/ r,
-/sys/class/*/ r,
-/sys/devices/** r,
-@{PROC}/*/mountinfo r,
-@{PROC}/swaps r,
-/sys/block/ r,
-/sys/devices/** r,
-/dev/bus/usb/ r,
-/dev/bus/usb/** r,
-/sys/bus/usb/devices/ r,
+# files in /sys pertaining to hardware
+/sys/{block,bus,class,devices}/{,**} r,
+
+# USB IDs
 /var/lib/usbutils/usb.ids r,
```

Comment:
Thanks for the cleanup! It just occurred to me that /var/lib/usbutils/usb.ids isn't going to exist on an all-snaps image or within the runtime environment of the snap so I think it doesn't make sense to have this rule?